### PR TITLE
Actions modal and icon UI update

### PR DESCRIPTION
**Pull request by joebudi**
## Description
 I updated the UI in several places, primarily the Automations section, to improve the design and experience.

I made styling changes that were not attached to our design system - specifically, the yellow and indigo within automations. These colors make the UI "pop." Happy to discuss this in more detail.

@deanhannigan I removed the left and right arrows on branches—I don't think they're needed. We can add them back if necessary.

### Automations

**Darkest**
CleanShot 2024-10-23 at 15 40 49@2x

**Light**
CleanShot 2024-10-23 at 15 43 03@2x

#### New Automation Modal
- removed unnecessary text
- Improved card spacing
- Added background colors

**Darkest**
CleanShot 2024-10-23 at 15 52 54@2x

**Light**
CleanShot 2024-10-23 at 15 44 14@2x

#### New Automation Action Modal
- spacing and layout changed - 3 columns
- added backgrounds to icons
- changed Icon colors
- Improved font size

_It would be beneficial to introduce a third category - Logic to this modal. I would also change Condition to "Continue If"_

**Darkest**

**Light**
CleanShot 2024-10-23 at 15 44 57@2x

### New Automation Card
- Card width reduced to accommodate more cards
- Single title - removed label.
- Icon background color added for trigger and action
- removed left and right options branching
- Label and fields split across two rows
- Spacing improved
- Colors fixed on light mode.

**Darkest**
CleanShot 2024-10-23 at 15 48 03@2x

### Tag
**Darkest**
CleanShot 2024-10-23 at 15 57 36@2x

**Light**

_CTA tag_
CleanShot 2024-10-23 at 15 45 56@2x

_Non-CTA tag_
CleanShot 2024-10-23 at 15 46 19@2x

### Alignment issue
CleanShot 2024-10-23 at 15 58 28@2x

_Add a small description in layman's terms of what this PR achieves. This will be used in the release notes._

I believe it improves the overall UX/UI

### CODE REVIEW

**Comment by aptkingston** [comment]
I think these changes look really good Joe 👌 

Only thing I spotted is that the looping icon needs updated to match the new icon styles:
image

I think the reduction in text here is welcome:
image
But do you reckon we could just remove it in general? We could put a warning icon up beside the activated toggle if the app is unpublished or something. I just think having this big warning in general looks not great and is very front and center.

I reckon we should also bring back the arrows for branches, because there are definitely cases where you'd want to reorder your branches, and the only option without that is to delete and recreate your conditions on every branch which would be quite tedious.

**Comment by joebudi** [comment]
I initially removed the warning from the new automation model. @PClmnt felt it was needed. I feel that if it is needed, this is not the place to show it - considering it's the first thing people see, and the action (publish) is the last thing users do.

Good spot with the looping icon. I wanted to create a 3rd category - Logic or something and provide a different color for them. If I was to make another change - I would change the shape of the trigger icon to a hexagon or something to make it really stand out - or the budibase logo arrow.

Branching arrows
There's a lot of power in those arrows - I don't like that they come before the warning - they look like a UI action for reordering branches. I think people will click them without reading the warning - at the bottom of the branch.
I would add a label above the branch to say something similar to the way Hubspot does it:
CleanShot 2024-10-25 at 09 49 57@2x
- first check this branch...
- then check branch 2 criteria...
- then check branch 3 criteria ...
- If none of the criteria are met, go here: (default "Conditions not met" branch added to every branch action)

CleanShot 2024-10-25 at 09 55 32@2x

Also... if we do add back the arrows, I'd probably add a more icon and place the delete, left, and right functionality in there.

**Comment by ben-budibase** [comment]
@joebudi @aptkingston 

https://www.figma.com/design/LZJRE6DsiGGCEjB7aqFYGx/Automations?node-id=704-5340&t=0pJ9dqYNWcX9ZPjL-1
Screenshot 2024-10-25 at 14 28 42

Joe and had a call earlier with version iterations on the automation branching. As you'll see I've amended the design to include Joe's latest additions as well as some of my own. These include having 'First check this branch', then 'then check this branch' at the top of the nodes. In addition, we've removed the info text and placed the branch movement buttons in a submenu with contextual labels, that way the user understands what they mean. 

Let me know your thoughts!

**Comment by joebudi** [comment]
@shogunpurple any updates/blockers on this one?
@ben-budibase might be best to get this live, and then talk about the Branching UX.
